writer.go

```diff
@@ -51,6 +51,17 @@ type Event struct {
 	Data  interface{} // Data value will be marshaled to JSON
 }
 
+// ResyncRequiredErrorEvent is the error event sent when a resync is required.
+func ResyncRequiredErrorEvent() Event {
+	return Event{
+		Event: "error",
+		Data: map[string]any{
+			"error": "resyncRequired",
+			"data":  map[string]any{},
+		},
+	}
+}
```

Comment:
I think we should have possibility to omit error data if not needed.

```suggestion
type ErrorEventData struct {
	Error string         `json:"error"`
	Data  map[string]any `json:"data,omitempty"`
}

func ErrorEvent(error string, data map[string]any) Event {
	return Event{
		Event: "error",
		Data: ErrorEventData{
			Error: error,
			Data:  data,
		},
	}
}

// ResyncRequiredErrorEvent is the error event sent when a resync is required.
func ResyncRequiredErrorEvent() Event {
	return ErrorEvent("resyncRequired", nil)
}
```